logstash-core/logstash-core.gemspec

```diff
@@ -86,7 +86,15 @@ Gem::Specification.new do |gem|
   # https://github.com/logstash-plugins/logstash-mixin-scheduler/blob/v1.0.1/lib/logstash/plugin_mixins/scheduler/rufus_impl.rb#L85=
   # and https://github.com/elastic/logstash/issues/13773
 
-  # TEMPORARY: racc-1.6.0 doesn't have JAVA counterpart (yet)
-  # SEE: https://github.com/ruby/racc/issues/172
-  gem.add_runtime_dependency "racc", "~> 1.5.2" #(Ruby license)
+  # TEMPORARY: delta between JRuby 9.4.2.0 and 9.4.3.0
+  gem.add_runtime_dependency "cgi", "~> 0.3.6"
+  gem.add_runtime_dependency "date", "~> 3.3.3"
+  gem.add_runtime_dependency "ffi", "~> 1.15.5"
+  gem.add_runtime_dependency "net-http", "~> 0.3.0"
+  gem.add_runtime_dependency "net-protocol", "~> 0.1.2"
+  gem.add_runtime_dependency "reline", "~> 0.3.5"
+  gem.add_runtime_dependency "ffi-binary-libfixposix", "~> 0.5.1.1"
+  gem.add_runtime_dependency "time", "~> 0.2.2"
+  gem.add_runtime_dependency "timeout", "~> 0.3.2"
+  gem.add_runtime_dependency "uri", "~> 0.12.1"
```

Comment:
I have run several configs, includes s3-input, http-input, tcp-input, etc.. and I didn't face any issues.
However, I am a bit curious about these declarations since I don't know internal details. 
Can I know based on what assumptions/references we added?

It seems to me scheduler doesn't have direct/indirect `net-http` dependency and test by removing it gives also successful result. Additionally, `net-http` seems internally includes `uri` (based on [this explanation](https://github.com/ruby/net-http#usage)) which may not explicitly define?

Reply:
Much of what was once the ruby standard library is now made available as separately-versioned gems, and ruby distributions like jruby now include "default gems" so that when a fix is made available in one component of the stdlib the user doesn't have to upgrade their whole ruby distribution — they can just install the individual gem and it will be used instead of its "default gems" counterpart.

We get these default gems "for free" by just using ruby (or jruby), so they aren't normally included in the explicit dependencies.

Jruby 9.4.3.0 includes a bunch of updates to these default gems over those shipped with the 9.4.2.0 that we are downgrading to. Declaring the updated versions here ensures that we have the most up-to-date versions including security fixes etc., and it also means that they become an explicit part of the dependency graph that is reflected in the generated lockfile. This is temporary, and can go away once the upstream issue is fixed.